### ISSUE TITLE
ios conformance app problem

### ISSUE THREAD

**Issue by ervanux** [fido-alliance/conformance-tool-issues#143]
By submitting this issue you are acknowledging that any information regarding this issue will be publicly available.

If you have privacy concerns, please email [conformance-tools@fidoalliance.org](mailto:conformance-tools@fidoalliance.org)

## What protocol and version of the protocol are you testing?

UAF v1.0

## What is your implementation class?

Client

## What is the version of the tool are you using?

v0.9.1

## Issue description

When running conformance tests app, successfully switching between my app and conformance app. But conformance app show below error and after 20 second jump next test. And same flow again.

> Error: Timeout of 2000ms exeeded. For async test and hooks ensure "done()" is called; if returning a Promise ensure it resolves...

here is an example of my uafoperation result url.

> FIDOConformaceToolsIonic://x-callback-url/UAF_OPERATION_RESULT?state=F538B8063AD146989AB59FC39C8C4F2A&json=eyJ0YWciOiJkWlIta1NjeFlwUHB3ZzFlT3hWdThnIiwiY2lwaGVydGV4dCI6Illsc3dIWDkzOERaeE16ZUQ4Sm0wT3ciLCJ1bnByb3RlY3RlZCI6IntcImFsZ1wiOlwiZGlyXCIsXCJlbmNcIjpcIkExMjhDQkMtSFMyNTZcIn0iLCJpdiI6IlcyRSJ9

**Comment by herrjemand** [comment]
@ervanux Replace `?` with `&` in `...UAF_OPERATION_RESULT?state=F538B8063AD146989AB59FC39C8...`

It's a known issue, and we are working to address it.

Regards. Yuriy

**Comment by ervanux** [comment]
@herrjemand it is working now. thank you.

**Comment by herrjemand** [comment]
For future reference regarding incorrect, the "correct" from ecosystem perspective is NOT to use ? but use &.

https://fidoalliance.org/specs/fido-uaf-v1.0-ps-20141208/fido-uaf-client-api-transport-v1.0-ps-20141208.html#ios-custom-url-api
This is very legacy issue that comes from the original typoe in example 7.1.1 Example 5: which used & instead of ?.

This was de-facto FIDO rule to use & instead of ? because all market implementations are implemented in such way.

This comment is just historical reference to the issue.

To summarize:

CORRECT (&)
```
FIDOConformaceToolsIonic://x-callback-url/UAF_OPERATION_RESULT&state=F538B80...
```

INCORRECT (?)

```
FIDOConformaceToolsIonic://x-callback-url/UAF_OPERATION_RESULT?state=F538B80...
```
